explorer/src/utils/index.ts

```diff
@@ -36,4 +45,20 @@ export function mixnodeToGridRow(arrayOfMixnodes: MixNodeResponse) {
     arr.push(formattedRow);
   })
   return arr;
+}
+
+export function gatewayToGridRow(arrayOfGateways: GatewayResponse) {
+  let arr: GatewayRows = [];
+  arrayOfGateways !== undefined && arrayOfGateways.forEach((eachRecord: GatewayResponseItem) => {
+    let formattedRow: GatewayRowType = {
+      id: eachRecord.owner,
+      owner: eachRecord.owner,
+      identity_key: eachRecord.gateway.identity_key || eachRecord.gateway.identity_key || '',
+      location: eachRecord?.gateway?.location || '',
+      bond: eachRecord.bond_amount.amount || 0,
+      host: eachRecord.gateway.host || ''
+    }
+    arr.push(formattedRow);
+  })
+  return arr;
```

Comment:
I think this might also work like this

```suggestion
export function gatewayToGridRow(arrayOfGateways: GatewayResponse) {
 return !arrayOfGateways ? [] : (
   arrayOfGateways.map(gw => ({
      id: gw.owner,
      owner: gw.owner,
      identity_key: gw.gateway.identity_key || "'',
      location: gw?.gateway?.location || '',
      bond: gw.bond_amount.amount || 0,
      host: gw.gateway.host || ''
   })
 )
}

```